src/cpp/src/tokenizer.cpp

```diff
@@ -86,11 +87,16 @@ class Tokenizer::TokenizerImpl {
 
     std::string m_chat_template = {};
 
-    void set_state_if_necessary(CircularBufferQueueElementGuard<ov::InferRequest>& infer_request_guard, bool add_special_tokens) {
+    void set_state_if_necessary(CircularBufferQueueElementGuard<ov::InferRequest>& infer_request_guard, const ov::AnyMap& params) {
+        bool add_special_tokens_flag = true;
+        bool skip_special_tokens_flag = false;
+        ov::genai::utils::read_anymap_param(params, add_special_tokens.name(), add_special_tokens_flag);
+        ov::genai::utils::read_anymap_param(params, skip_special_tokens.name(), skip_special_tokens_flag);
+
         // If user requested add_special_tokens mode different from the current one,
         // need to set state variable.
         // If requested mode matches the stored state set, then don't touch states.
-        if (add_special_tokens == m_add_special_tokens) {
+        if (add_special_tokens_flag == m_add_special_tokens && skip_special_tokens_flag == m_skip_special_tokens) {
             return;
         }
         if (m_older_than_24_5) {
```

Comment:
does m_skip_special_tokens work with IRs produced by 2024.4 and older?

Reply:
if this PR depends on https://github.com/openvinotoolkit/openvino_tokenizers/pull/325, than we need to add condition for 2025.0 for `skip_special_tokens`?

Reply:
about  2025.0 condition, need to discuss with @apaniukov

Reply:
We don't need condition for 2025.0. If IR is older then mechanism from this genai PR simply will not be activated (`MakeVocabDecoderSatateful` pass simply will not be matched) and the results will be the same as with older genai.